### DISCUSSION TITLE
想问下如何使播放器自动全屏？

### DISCUSSION THREAD

**Discussion by Tkey20** [bytedance/xgplayer#876] [Q&A] [answered]
这是我下载的版本
"xgplayer": "^2.17.1",
"xgplayer-flv.js": "^2.3.0"

现在就是每次调用全屏这个方法的时候 this.player.getFullscreen(this.player.root);
浏览器总是会警告
![image](https://user-images.githubusercontent.com/67895563/219571630-7f028160-5a2a-4091-ab9a-4dcbdcf649e5.png)

想问下有没有办法解决这个问题的

**Comment by gemxx** [comment, marked as answer]
必需通过用户交互才能全屏，可以参考[requestFullScreen security](https://developer.mozilla.org/en-US/docs/Web/API/Element/requestFullScreen#security) 部分进行了解

> [Transient user activation](https://developer.mozilla.org/en-US/docs/Web/Security/User_activation) is required. The user has to interact with the page or a UI element in order for this feature to work.
